src/index.d.ts

```diff
@@ -0,0 +1,19 @@
+type StyleObject = {
+  [key: string]: unknown;
```

Comment:
Hi! Not sure about the unknown type, seems a little restrictive for this use case, I would rather go with `any` here. Or maybe don't validate the StyleObject type at all, using `object` instead (?).

Reply:
Actually, the `unknow` type is the safe counterpart of `any`. It's a safer type than any, and a better practice. And the `object` type refers to all the non-primitive values (like String, null, undefined), but it doesn't allow us to access inner properties, so typing the object map adding a string type to the key is a better approach.

I say this after working with typescript for 3 years +, but it's your call.

[TS Playground](https://www.typescriptlang.org/play?#code/FAMwrgdgxgLglgewgAgCYIMoILYFMYAWcEA5sgBQIBGAVgFzLU26wCUyA3sMj8gE74wfFEwB0EAIZ4A3MAC+wUJFiIU6LHkLESAJgpMGXXsgDaAa1wBPBgGcYfbQF0GkMxAQB3CPPZHeAmCERWnEpXFkFJWh4JDRMHHwiUgBmfVpDbl5zK1t7JwYJCEsfTkyeAKDGEMkZeUzgKCQ7KppkAF5S4xrcBgByABVcO1664HUErVJKWlZZcc0k3WmaWbH4he1k5dmgA)
[Unknown Type](https://mariusschulz.com/blog/the-unknown-type-in-typescript)

Reply:
Ok, I understand your point, yes object seems a bad idea; I am not an expert in typescript; what I meant is that I am not sure if forcing the user to cast the value is the way to go, it could be counterproductive.

```
function doSomething2 (obj: {
    [key: string]: unknown
}) {
    return obj.name.toString(); // Error
}

function doSomething3 (obj: {
    [key: string]: any
}) {
    return obj.name.toString(); // Ok
}
```

Reply:
Just to explain it a little more:

This example is exactly where the `unknown` type provides a safer usage. You can do anything with `any`, so there is no type safety with it, that's why TS has a strict mode where you block the usage of `any` (this could be an issue for the users). On the other hand, the `unknown` needs a cast before usage, so you need to provide the correct type to it before using it, providing a better type safety.

```
function doSomething2 (obj: {
    [key: string]: unknown
}) {
    return (obj.name as string).toString(); // ok
}
```

The thing is if something is ʻunknown`, how do we can be sure it will have the `toString ()` method? We don't, and if we know it for sure, we simply cast it. In both cases, TS will provide the correct warnings and type safety for us.

But I agree, giving the correct type could be counterproductive for beginners with TS, it's a trade-off. No problem for me if you want to go with any 😄. We can always change it later if there is any kind of issue with one of both uses.

Reply:
Great, seems good to me; either way, I don't think this casting thing is going to be something usual. I'll merge this ASAP. Thanks for the explanation and for your contribution!